<launch>
	<!-- vim: set ft=xml noet : -->
	<!-- example launch script for PX4 based FCU's -->

	<arg name="fcu_url" default="/dev/ttyUSB0:921600" />
	<arg name="gcs_url" default="" />
	<arg name="tgt_system" default="1" />
	<arg name="tgt_component" default="1" />
	<arg name="log_output" default="screen" />

	<include file="$(find mavros)/launch/node.launch">
		<arg name="pluginlists_yaml" value="$(find mavros)/launch/px4_pluginlists.yaml" />
		<arg name="config_yaml" value="$(find mavros)/launch/px4_config.yaml" />

		<arg name="fcu_url" value="$(arg fcu_url)" />
		<arg name="gcs_url" value="$(arg gcs_url)" />
		<arg name="tgt_system" value="$(arg tgt_system)" />
		<arg name="tgt_component" value="$(arg tgt_component)" />
		<arg name="log_output" value="$(arg log_output)" />
	</include>

	<!-- Waypoint Manager -->
    <node name="waypoint_manager_mavros" pkg="ibvs_sim" type="ibvs_state_machine.py" output="screen">
      <param name ="mode" value="mavros" />

      <param name="wp_threshold" value="0.5" />
      <param name="u_max_inner" value="0.5" />
      <param name="v_max_inner" value="0.5" />
      <param name="w_max_inner" value="0.3" />

      <param name="inner_error_condition" value="True" />

      <remap from="estimate" to="ground_truth/odometry/NED"/>
      <!-- <remap from="estimate" to="estimate"/> -->

      <remap from="/target_position" to="/boat/odometry/NED"/>
    </node>

</launch>